tiledb/sm/c_api/tiledb.h

```diff
@@ -3848,6 +3849,31 @@ TILEDB_EXPORT int32_t tiledb_query_set_condition(
     tiledb_query_t* query,
     const tiledb_query_condition_t* cond);
 
+/**
+ * Sets the datatype for attribute/dimentsion to be written or read.
```

Comment:
> I don't think we should implement this at the filter pipeline level. Access to the offsets for var length conversions isn't possible in the pipeline paths. My concern is this will have to be resigned for var length.
> 
> Can you evaluate implementing this higher in the stack, like how the offset conversation from bytes to elements or offset 64bit to 32 bit is handled?

Yes, I will evaluate this more. I was thinking about how to implement var length conversions using the filter which seems not trivial. We might need to change the design

Reply:
@bdeng-xt another option besides the offset conversions is to take a look at the query conditions. The query conditions operate at the right level of the code, they have full access to var length and null data. Its likely we can leverage the framework build for query conditions and make it more generic to handle additional items such as these datatype conversions. I'm not suggesting to add the conversions to the conditions, just how we implemented conditions could also be used here perhaps.

Reply:
We assumed the reading/writing buffer to be the same type as that in the array schema. The original design with a pipeline filter will make the implementation more difficult for var length conversions and writing, although it is possible. After I investigated our query condition codes, I decided to redesign the data type conversions.  This pull request will be closed and a new branch for data type conversion will be created,